flow/project.py

```diff
@@ -1619,13 +1619,22 @@ def _print_progress(x):
 
         with self._potentially_buffered():
             try:
-                with contextlib.closing(ThreadPool()) as pool:
-                    _map = map if no_parallelize else pool.imap
-                    # First attempt at parallelized status determination.
-                    # This may fail on systems that don't allow threads.
-                    return list(tqdm(
-                        iterable=_map(_get_job_status, jobs),
-                        desc="Collecting job status info", total=len(jobs), file=err))
+                if execute_ThreadPool:
+                    with contextlib.closing(ThreadPool()) as pool:
+                        _map = map if no_parallelize else pool.imap
+                        # First attempt at parallelized status determination.
+                        # This may fail on systems that don't allow threads.
+                        return list(tqdm(
+                            iterable=_map(_get_job_status, jobs),
+                            desc="Collecting job status info", total=len(jobs), file=err))
+                else:
+                    with contextlib.closing(Pool()) as pool:
+                        _map = map if no_parallelize else pool.imap
+                        # First attempt at parallelized status determination.
+                        # This may fail on systems that don't allow threads.
+                        return list(tqdm(
+                            iterable=_map(_get_job_status, jobs),
+                            desc="Collecting job status info", total=len(jobs), file=err))
```

Comment:
The code in the if and else blocks are the same (right?). Can you refactor to get rid of that duplication. Something like

```python
if execute_ThreadPool:
    pool_ = ThreadPool()
else:
    pool_ = Pool()
with contextlib.closing(pool_) as pool:
    # do stuff
````

Reply:
Will do and push the changes